pkg/scheduler/schedule_one_test.go

```diff
@@ -852,6 +868,17 @@ func TestSchedulerScheduleOne(t *testing.T) {
 				if diff := cmp.Diff(item.expectBind, gotBinding); diff != "" {
 					t.Errorf("got binding diff (-want, +got): %s", diff)
 				}
+				if item.eventReason == "Scheduled" {
```

Comment:
Instead of adding a new wrapping function `podWithCondition`, and update a lot manually in L846-L855.
I think we can just simply iterate pod's conditions when `item.eventReason == "Scheduled"` and check if the correct condition `{ Type:   v1.PodScheduled, Status: v1.ConditionTrue}` is contained?

Also, can we also add a check for `managedFields`(see if it's set correctly) as this is the origin purpose of the issue.

Reply:
Ok, got it, So we actually have already compared the condition.
Then yes, we should keep the `podWithCondition`, but still we don't have to modify the `LastTransitionTime` manually, instead we can just ignore them by using `cmp.DIff` instead of `reflect.DeepEqual`

```
	if diff := cmp.Diff(pod2, pod1, cmpopts.IgnoreFields(v1.PodCondition{}, "LastTransitionTime")); diff != "" {
		fmt.Println(diff)
	}
```

> Also, can you point me to a place where managedFields testing is done scheduler ?

I'm also not familiar with this. Will take a look later.

Reply:
> @AxeZhan I have modified an existing e2e test to test the managedFields condition.

That's great, can we use `cmp.Diff` here to remove the modify to `LastTransitionTime` though?